[BENCHMARK] Run parameter benchmarks with different input data and update them.

Irallia opened the pull request:
# min_var_length = 30 (remains)

old  value | new value
:---------:|:----------:
![min_var_length results all](https://user-images.githubusercontent.com/5746285/169673240-a1168e8e-8ef6-4cea-9a7b-7187186d14a8.png) | ![min_var_length results all](https://user-images.githubusercontent.com/5746285/169672719-d5475032-b67e-4d92-8e1f-c04fbb5cbfc3.png)
![min_var_length results all](https://user-images.githubusercontent.com/5746285/169673300-bbeb3cc1-a162-4d50-8e32-d4efee609c04.png) | ![min_var_length results all](https://user-images.githubusercontent.com/5746285/169672781-607d8adf-bb72-4fee-9845-714fc3bf5f35.png)
![min_var_length results all](https://user-images.githubusercontent.com/5746285/169673344-e4bb715c-7139-4624-a53d-5960fdd6c7cf.png) | ![min_var_length results all](https://user-images.githubusercontent.com/5746285/169672791-edb3803d-96b4-4c6d-a610-c162ac4c14d8.png)
![min_var_length results all](https://user-images.githubusercontent.com/5746285/169673373-66a213bf-3210-419c-a28d-287970478f6c.png) | ![min_var_length results all](https://user-images.githubusercontent.com/5746285/169672796-bf6ec390-c63a-418e-ac49-bc1cb3729dae.png)
![min_var_length results all](https://user-images.githubusercontent.com/5746285/169673401-21ececc3-f4d5-473f-98b3-55f1a698c635.png) | ![min_var_length results all](https://user-images.githubusercontent.com/5746285/169672671-b32c115a-1873-492d-a4ae-75bcb4b87ddc.png)

# max_var_length = 10.000 (changed from 100.000)

old  value | new value
:---------:|:----------:
![max_var_length results all](https://user-images.githubusercontent.com/5746285/169673239-ca98be64-8aa2-406c-a50a-858f1dd15e97.png) | ![max_var_length results all](https://user-images.githubusercontent.com/5746285/169672881-8a05d330-13c0-4aba-b2ea-f64eeae93571.png)
![max_var_length results all](https://user-images.githubusercontent.com/5746285/169673298-6752d4e3-2d1f-4214-bc73-f1c9415c90a7.png) | ![max_var_length results all](https://user-images.githubusercontent.com/5746285/169672984-274c86af-2d10-4c78-bb24-5e3ea6121834.png)
![max_var_length results all](https://user-images.githubusercontent.com/5746285/169673343-4b79cdec-a8f0-408a-bee0-86c18b266fcf.png) | ![max_var_length results all](https://user-images.githubusercontent.com/5746285/169673079-0a2f9a68-91fb-416e-8bbf-838f96ef9f81.png)
![max_var_length results all](https://user-images.githubusercontent.com/5746285/169673372-54169c1f-0b88-4903-9ba5-dde2d5d96770.png) | ![max_var_length results all](https://user-images.githubusercontent.com/5746285/169672869-ccb24c24-dabc-4b93-842c-cf72499f9fc5.png)
![max_var_length results all](https://user-images.githubusercontent.com/5746285/169673399-d5153769-b121-42b7-9900-a7adfc2183dc.png) | ![max_var_length results all](https://user-images.githubusercontent.com/5746285/169673206-8f4c930d-ee9e-425a-a667-303e50c7a6c6.png)

# max_tol_inserted_length = 50 (remains)

old  value | new value
:---------:|:----------:
![max_tol_inserted_length results all](https://user-images.githubusercontent.com/5746285/169673238-085e399f-4363-406c-8d3f-4ef45831bb4f.png) | ![max_tol_inserted_length results all](https://user-images.githubusercontent.com/5746285/169672893-a2bbcf7d-f24c-438b-ad7b-88df9dc2b2cb.png)
![max_tol_inserted_length results all](https://user-images.githubusercontent.com/5746285/169673297-acea9666-6215-40cc-9ee7-0d2bdb6a8c11.png) | ![max_tol_inserted_length results all](https://user-images.githubusercontent.com/5746285/169672982-025e4489-d658-4d8c-aea2-156f3b0a047a.png)
![max_tol_inserted_length results all](https://user-images.githubusercontent.com/5746285/169673341-8481354c-f038-4ee3-931e-e4b94c961068.png) | ![max_tol_inserted_length results all](https://user-images.githubusercontent.com/5746285/169673077-16b7982c-fae3-449b-9c8c-0e2f88f04101.png)
![max_tol_inserted_length results all](https://user-images.githubusercontent.com/5746285/169673371-1635d25d-647b-4106-8858-80d2e8d84409.png) | ![max_tol_inserted_length results all](https://user-images.githubusercontent.com/5746285/169673119-83196de0-1ab7-480d-836e-69e5ddfe830d.png)
![max_tol_inserted_length results all](https://user-images.githubusercontent.com/5746285/169673398-b149b9fe-8a7f-42f9-945c-f8196b75201d.png) | ![max_tol_inserted_length results all](https://user-images.githubusercontent.com/5746285/169673205-d160b75c-9fb5-4754-9f50-b7bc545c46f9.png)

# max_tol_deleted_length = 50 (remains)

old  value | new value
:---------:|:----------:
![max_tol_deleted_length results all](https://user-images.githubusercontent.com/5746285/169673237-144229f9-c551-404f-b47d-dcd4ddaf968c.png) | ![max_tol_deleted_length results all](https://user-images.githubusercontent.com/5746285/169672892-8337ddb7-c5d1-4c7b-a79d-66d1af4cfb9c.png)
![max_tol_deleted_length results all](https://user-images.githubusercontent.com/5746285/169673296-1873a1d6-0362-4bb5-bbfe-66d6094a53d7.png) | ![max_tol_deleted_length results all](https://user-images.githubusercontent.com/5746285/169672980-44a43923-8364-4e3a-951b-f07c2a70da3b.png)
![max_tol_deleted_length results all](https://user-images.githubusercontent.com/5746285/169673340-f0bec0ac-3f45-455f-a767-eeb79f47d2a0.png) | ![max_tol_deleted_length results all](https://user-images.githubusercontent.com/5746285/169673075-70ee7e4b-5e0d-4c45-a3b0-097300724cf0.png)
![max_tol_deleted_length results all](https://user-images.githubusercontent.com/5746285/169673370-fdd5a46d-727c-4c6d-834a-485aa7d5b6c3.png) | ![max_tol_deleted_length results all](https://user-images.githubusercontent.com/5746285/169673118-761bbeaf-ca99-4d98-86c7-4479c3396a64.png)
![max_tol_deleted_length results all](https://user-images.githubusercontent.com/5746285/169673397-d16a8a7b-bcab-45fa-bfeb-22fd8072a088.png) | ![max_tol_deleted_length results all](https://user-images.githubusercontent.com/5746285/169673204-18326933-e580-48cb-a731-401d5fe6fcd8.png)

# max_overlap = 50 (changed from 10)

old  value | new value
:---------:|:----------:
![max_overlap results all](https://user-images.githubusercontent.com/5746285/169673236-100a501b-93eb-4193-9d79-6f5d57053e14.png) | ![max_overlap results all](https://user-images.githubusercontent.com/5746285/169672891-2606db3b-0ab8-4f87-8771-6d07cd61a7e6.png)
![max_overlap results all](https://user-images.githubusercontent.com/5746285/169673295-e18ebdf8-0571-4a3c-9806-0dd4ab9f3971.png) | ![max_overlap results all](https://user-images.githubusercontent.com/5746285/169672978-98659f0e-e2fa-42d6-9bcb-b0550640e6b0.png)
![max_overlap results all](https://user-images.githubusercontent.com/5746285/169673339-965d8515-4586-4e97-8a04-ccc8d27a4c1c.png) | ![max_overlap results all](https://user-images.githubusercontent.com/5746285/169673074-107bffb5-eb88-4103-8fb2-096423979c5e.png)
![max_overlap results all](https://user-images.githubusercontent.com/5746285/169673369-7558389d-e727-4df8-9b43-5ee09f4ea6f5.png) | ![max_overlap results all](https://user-images.githubusercontent.com/5746285/169673117-48c45764-8cd2-43d5-9531-03da3df95633.png)
![max_overlap results all](https://user-images.githubusercontent.com/5746285/169673395-7cec6c36-f420-4710-9739-2e4b2727e9d6.png) | ![max_overlap results all](https://user-images.githubusercontent.com/5746285/169673202-ae872a4c-f6a5-4e7b-8645-74f1f0d5ea3f.png)

# partition_max_distance = 50 (changed from 1.000)

old  value | new value
:---------:|:----------:
![partition_max_distance results all](https://user-images.githubusercontent.com/5746285/169673241-8487d001-b5db-4124-ae70-854969595fab.png) | ![partition_max_distance results all](https://user-images.githubusercontent.com/5746285/169672897-3f58935c-d0b5-405b-a653-5da1da38af79.png)
![partition_max_distance results all](https://user-images.githubusercontent.com/5746285/169673301-2cc79f4f-427f-42e9-a463-63c77ef7c097.png) | ![partition_max_distance results all](https://user-images.githubusercontent.com/5746285/169672986-da0987af-9908-4296-acda-e0a88473bd95.png)
![partition_max_distance results all](https://user-images.githubusercontent.com/5746285/169673345-df2880b5-259e-495b-8cc6-e67a1b1da53d.png) | ![partition_max_distance results all](https://user-images.githubusercontent.com/5746285/169673080-5f65f23e-8756-4ade-a142-c56c2df08771.png)
![partition_max_distance results all](https://user-images.githubusercontent.com/5746285/169673374-290124ce-0bc8-400e-9bf3-1dd381cc570f.png) | ![partition_max_distance results all](https://user-images.githubusercontent.com/5746285/169673122-44ca9368-02f6-429e-8054-8aeab1c4dfd0.png)
![partition_max_distance results all](https://user-images.githubusercontent.com/5746285/169673402-5925c258-f503-4acf-b5df-35fd293ac55e.png) | ![partition_max_distance results all](https://user-images.githubusercontent.com/5746285/169673207-d51efb02-005b-46c1-b358-cf37826effbd.png)

# hierarchical_clustering_cutoff = 0.3 (changed from 0.5)

old  value | new value
:---------:|:----------:
![hierarchical_clustering_cutoff results all](https://user-images.githubusercontent.com/5746285/169673235-fe30f24d-e7de-48a0-8a01-45b2306c9a2f.png) | ![hierarchical_clustering_cutoff results all](https://user-images.githubusercontent.com/5746285/169672889-904aa863-e764-440a-894a-2a82d9847c67.png)
![hierarchical_clustering_cutoff results all](https://user-images.githubusercontent.com/5746285/169673294-d03ab5e7-5aa8-4ad6-b210-8fabd670de4d.png) | ![hierarchical_clustering_cutoff results all](https://user-images.githubusercontent.com/5746285/169672977-c49416a4-6b13-4532-9972-23519b55f49a.png)
![hierarchical_clustering_cutoff results all](https://user-images.githubusercontent.com/5746285/169673338-194e65b6-a6e1-4743-ad64-afce0c2539bd.png) | ![hierarchical_clustering_cutoff results all](https://user-images.githubusercontent.com/5746285/169673072-c9edc220-80af-4408-b9bf-95fe9f11a9f7.png)
![hierarchical_clustering_cutoff results all](https://user-images.githubusercontent.com/5746285/169673368-a29a8fe9-7479-485e-b458-3168329896e4.png) | ![hierarchical_clustering_cutoff results all](https://user-images.githubusercontent.com/5746285/169673116-ebe6c1a1-3db0-4760-98b7-66f3f1cbad57.png) 
![hierarchical_clustering_cutoff results all](https://user-images.githubusercontent.com/5746285/169673393-b3c6d1af-c49e-449f-89bb-61b23590e8cf.png) | ![hierarchical_clustering_cutoff results all](https://user-images.githubusercontent.com/5746285/169673201-be1cc7a1-1a2c-4994-b886-0bffd260c9c1.png)

Irallia commented:
# iGenVar only - new Plots:

old  values | new values
:----------:|:-----------:
![iGenVar_only-results all](https://user-images.githubusercontent.com/5746285/169673551-774f2652-77c7-4a0f-9400-885953328550.png) | ![iGenVar_only-results all](https://user-images.githubusercontent.com/5746285/169673562-6f12efa8-6665-41bd-bce6-431e0bd5f789.png)
![iGenVar_only-results DUP_as_INS all](https://user-images.githubusercontent.com/5746285/169673552-f4b90900-9906-429c-99d8-a4d678a86067.png) | ![iGenVar_only-results DUP_as_INS all](https://user-images.githubusercontent.com/5746285/169673564-6e94f4c2-cd8d-4e52-92d2-dcae1194c789.png)

joergi-w commented on test/benchmark/parameter_benchmarks/Snakefile:
```diff
@@ -83,35 +115,49 @@ rule reformat_truvari_results:
+                                    config["quality_ranges"]["iGenVar"]["step"]))),
+        input7 = expand("results/parameter_benchmarks/{{dataset}}/truvari/hierarchical_clustering_cutoff/{parameter_value}_min_qual_{min_qual}/pr_rec.txt",
+                parameter_value= [0.1, 0.2, 0.3, 0.4],          # default: 0.3
+                min_qual=list(range(config["quality_ranges"]["iGenVar"]["from"],
+                                    config["quality_ranges"]["iGenVar"]["to"],
+                                    config["quality_ranges"]["iGenVar"]["step"])))
```
This is copied for in each input – is there a way to pre-define this?

joergi-w commented on test/benchmark/parameter_benchmarks/Snakefile:
```diff
@@ -2,38 +2,69 @@ configfile: "Repos/iGenVar/test/benchmark/config/caller_comparison_config.yaml"
+                dataset=["Illumina_Paired_End", "Illumina_Mate_Pair", "MtSinai_PacBio", "PacBio_CCS", "10X_Genomics"],
+                parameter_name="partition_max_distance"),
+        expand("results/parameter_benchmarks/{dataset}/plots/{parameter_name}.results.all.png",
+                dataset=["Illumina_Paired_End", "Illumina_Mate_Pair", "MtSinai_PacBio", "PacBio_CCS", "10X_Genomics"],
```
the dataset is also always the same

Irallia replied:
Good point, this appears in every benchmark workflow. So I think I will try to make it shorter in another PR.

joergi-w reviewed (approved):
Looks good! 😄 
Two ideas for the code:

joshuak94 reviewed (approved):
Looks good! How did you come up with the new values?

Irallia commented:
> Looks good! How did you come up with the new values?

With exactly this PR. So far I have tested iGenvar only on one dataset regarding the parameters, now I have added others. And then I changed only one parameter at a time and saw how our result behaves. With some back and forth I have now ended up with the dafault values and plots created here. On the plots you can see that larger or smaller values would not give any improvement.
